### DISCUSSION TITLE
HA with docker, how to add this addon?

### DISCUSSION THREAD

**Discussion by antgustech** [homeassistant-apps/app-cloudflared#329] [Q&A]
Addons are not available in HA running docker. Has anyone managed to get this running as a separate container somehow?

**Comment by brenner-tobias** [comment]
This is unfortunately not simply possible.
But, if you are using Docker, I suggest to simply use the Cloudflared Docker Container with the Managed Tunnel, as described [here](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/installation/#docker).

**Comment by antgustech** [comment]
Sadly, no arm32 bit is supported.

**Reply by CNvdH** [reply]
I've managed to get it working using this cloudflared on arm64 image: https://hub.docker.com/r/milgradesec/cloudflared

follow these steps to set up a cloudflared tunnel from a docker container (just use milgradesec/cloudflared instead of cloudflare/cloudflared): https://www.youtube.com/watch?v=GCQHulz5fjY&list=PLY4sFY6dmLqwQrcEHPnqVi8EayhdP7yLK&index=3

Then follow these steps to get it to work with HA: https://www.youtube.com/watch?v=cInnpVEW50k
